cirq-google/cirq_google/serialization/circuit_serializer.py

```diff
@@ -114,6 +134,13 @@ def _serialize_circuit(
                         raw_constants=raw_constants,
                     )
 
+            if self.use_constants_table_for_moments:
+                # Add this moment to the constants table
+                constants.append(v2.program_pb2.Constant(moment_value=moment_proto))
+                raw_constants[moment] = len(constants) - 1
```

Comment:
what is the constant at 0? ... it seems that the condition in the deserialization logic assume that the moment indicies are positive ... but they are just and index which could be 0, but since the tests pass there seem to be a constant at 0 ... is this by design? and is it always the case?

Reply:
I am not sure I understand this comment.  raw_constants just tells you what the index in the constants table is for each object.  It's a repeated proto list, so the index starts at zero.  We append to the table in line 139, so there is at least one element, and then we set the index for this element to `len(constants) - 1` which is the last index in the list (which is the one we just added).

Reply:
Oh, I see!  Yeah, I think that is a serious issue.  I changed the field to optional and now check presence rather than truthiness.